python/ray/tune/logger.py

```diff
@@ -181,10 +180,12 @@ def on_result(self, result):
         self._file_writer.flush()
 
     def flush(self):
-        self._file_writer.flush()
+        if hasattr(self, "_file_writer"):
```

Comment:
can we not use `hasattr` and instead set a class variable `_file_writer = None`?

Reply:
updated

Reply:
thanks.